File: Problem_2.java
public class Solution extends Relation{
    public int findCelebrity(int n){
        int celeb = 0;
        for (int i=1; i<n; i++){
            if (knows(celeb, i)){
                celeb = i;
            }
        }
        // potential celeb
        for (int i = 0; i < n; i++){
            if (i == celeb) continue;
            if (i < celeb){
                if (!knows(i, celeb) || knows(celeb, i)) return -1;
            } else{
                if (!knows(i, celeb)) return -1;
            }
        }
        return celeb;
    }
}

// Time Complexity: O(n)
// Space Complexity: O(1)

File: problem2.cpp
/*
// Time Complexity : O(N)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach
If a knows b a is not a celebrity jump to b. If a doest know b then b is not a celebrity. 
Then for celebrity make sure it is a celebrity by checking all the knows.
*/

#include<iostream>

bool knows(int a,int b); //API

/* The knows API is defined for you.
      bool knows(int a, int b); */

class Solution {
public:
    int findCelebrity(int n) {
        int i{},j{},res{};
        while(j<n){
            if(i==j){
                ++j;
            }
            else{
                if(knows(i,j)){
                    i=j;
                }
                else{
                    ++j;
                }
            }
        }
        res = i;
        for(int i{};i<n;++i){
            if(i == res) continue;
            if(!knows(i,res) || knows(res,i)) return -1;
        }
        return res;
    }
};


